test/libsolidity/semanticTests/functionCall/gas_and_value_basic.sol

```diff
@@ -38,7 +38,7 @@ contract test {
 }
 // ----
 // constructor(), 20 wei ->
-// gas irOptimized: 252642
+// gas irOptimized: 260928
```

Comment:
The change does affect some gas costs but very minimally. The only test where difference is somewhat significant is this one (and the similar  `gas_and_value_brace_syntax.sol`) where it's ~3%.

Reply:
Have you looked at the diff of optimized yul? Not sure how well it'd compare, though...

Reply:
It's also generally a problem that the code size metric used for this is merely a rather coarse estimate in itself...

Reply:
Not yet. I'll check this.

Reply:
Given the very few gas changes in this PR - could it be that the first pass retains the same code size, whilst actually optimizing the code - i.e. altering it, and then only reduces it during the second iteration?

Reply:
This is the relevant part of the diff:

```diff
                                 revert(0, 0x24)
                             }
                             mstore(_2, _1)
+                            _4 := 0
                         }
                         let memPos := mload(_2)
                         mstore(memPos, 0x01)
                         return(memPos, 32)
                     }
                     case 0x96dfcbea {
-                        if callvalue() { revert(0, 0) }
-                        if slt(add(calldatasize(), not(3)), 0) { revert(0, 0) }
-                        let value_1 := and(sload(0), sub(shl(160, 1), 1))
-                        let _5 := mload(_2)
-                        mstore(_5, shl(228, 0x0f963393))
-                        let _6 := call(gas(), value_1, 0, _5, _3, _5, 32)
-                        if iszero(_6)
+                        if callvalue() { revert(_4, _4) }
+                        if slt(add(calldatasize(), not(3)), _4) { revert(_4, _4) }
+                        let value_1 := and(sload(_4), sub(shl(160, 1), 1))
+                        let _6 := mload(_2)
+                        mstore(_6, shl(228, 0x0f963393))
+                        let _7 := call(gas(), value_1, _4, _6, _3, _6, 32)
+                        if iszero(_7)
                         {
                             let pos_1 := mload(_2)
-                            returndatacopy(pos_1, 0, returndatasize())
+                            returndatacopy(pos_1, _4, returndatasize())
                             revert(pos_1, returndatasize())
                         }
-                        let expr := 0
-                        if _6
+                        let expr := _4
+                        if _7
                         {
-                            let _7 := 32
-                            if gt(32, returndatasize()) { _7 := returndatasize() }
-                            finalize_allocation(_5, _7)
-                            if slt(sub(add(_5, _7), _5), 32) { revert(0, 0) }
-                            let value_2 := mload(_5)
-                            if iszero(eq(value_2, iszero(iszero(value_2)))) { revert(0, 0) }
+                            let _8 := 32
+                            if gt(32, returndatasize()) { _8 := returndatasize() }
+                            finalize_allocation(_6, _8)
+                            if slt(sub(add(_6, _8), _6), 32) { revert(_4, _4) }
+                            let value_2 := mload(_6)
+                            if iszero(eq(value_2, iszero(iszero(value_2)))) { revert(_4, _4) }
                             expr := value_2
                         }
                         let var_myBal := selfbalance()
```

I.e. it seems that a bunch of `0` literals used to be materialized and now remain as variables.

Which is odd - I wanted to suggest maybe adding another `T` to the sequence but it's right there in the cleanup : `fDnTOcmu`.